umap/static/umap/js/modules/data/layer.js

```diff
@@ -802,13 +802,12 @@ export class DataLayer {
       this
     )
     if (this.umap_id) {
-      const download = DomUtil.createLink(
-        'button umap-download',
-        advancedButtons,
-        translate('Download'),
-        this._dataUrl(),
-        '_blank'
-      )
+      const filename = Utils.slugify(this.options.name)
```

Comment:
Probably miss an extension?

Reply:
Meeh, good point!

Reply:
humm, actually, the downloaded file does have an extension. I guess because we are calling the back office, which set one. So does the `download` attribute still make sense ?

Reply:
No, it's not the backend that sets this extension. Can it be firefox adding it, as the backend sets the content-type to `application/geo+json ?

Reply:
The name of the layer ;)

![image](https://github.com/user-attachments/assets/567450b5-110b-46bb-b521-e851a21cac68)